src/types/Color.ts

```diff
@@ -0,0 +1,83 @@
+import * as _ from 'lodash'
+
+export const Color: SBVRType<number, RGBA> = {
+	types: {
+		postgres: 'INTEGER',
+		mysql: 'INTEGER',
+		websql: 'INTEGER',
+		odata: {
+			name: 'Self.Color',
+			complexType: `
+				<ComplexType Name="Color">
+					<Property Name="r" Nullable="false" Type="Edm.Int8"/>
+					<Property Name="g" Nullable="false" Type="Edm.Int8"/>
+					<Property Name="b" Nullable="false" Type="Edm.Int8"/>
+					<Property Name="a" Nullable="false" Type="Edm.Int8"/>
+				</ComplexType>
+			`
+			}
+	},
+
+	nativeProperties: {
+		has: {
+			'Red Component': (from:string) => ['BitwiseAnd', ['BitwiseShiftRight', from, 16], 255],
+			'Green Component': (from:string) => ['BitwiseAnd', ['BitwiseShiftRight', from, 8], 255],
+			'Blue Component': (from:string) => ['BitwiseShiftRight', from, 255],
+			'Alpha Component': (from:string) => ['BitwiseAnd', ['BitwiseShiftRight', from, 24], 255],
+		}
+	},
+
+	fetchProcessing: (data, callback) => {
+		callback(null, {
+			r: (data >> 16) & 0xFF,
+			g: (data >> 8) & 0xFF,
+			b: data & 0xFF,
+			a: (data >> 24) & 0xFF,
+		})
+	},
+
+	validate: (value, required, callback) => {
+		let processedValue = 0
+		if (!_.isObject(value)) {
+			processedValue = _.parseInt(value)
+
+			if (_.isNaN(processedValue)) {
+				return callback(`is neither an integer or color object: ${value}`)
+			}
+	 	} else {
+			for (const component in value) {
+				if (!value.hasOwnProperty(component)) continue;
```

Comment:
Can you use braces for this (I know I didn't because I copy/pasted from the coffee-script output, but seeing it mixed in with the other code makes the difference noticeable)